mita/openstack.py

```diff
@@ -135,17 +135,29 @@ def get_driver_v2(self):
         """
         return self.driver_v2 if self.driver_v2 else self.get_driver(api_version="2.2")
 
-    def _get_image(self) -> NodeImage:
-        """Return the glance image reference."""
-        try:
-            return [
-                i for i in self.driver_v2.list_images() if i.name == self.image_name
-            ][0]
-        except IndexError:
-            raise ResourceNotFound("Image {} not found".format(self.image_name))
-        except BaseException as be:  # noqa
-            logger.error(be)
-            raise OpenStackDriverError("Encountered an unknown exception.")
+    def _get_image(self, name: Optional[str] = None) -> NodeImage:
+        """
+        Return the NodeImage instance using the image_name or the provided name.
+
+        Args:
+            name: The name of the image.
+
+        Return:
+            NodeImage instance that is referenced by the name.
+
+        Raises:
+            ResourceNotFound - when no resource was found.
+        """
+        name = self.image_name if name is None else name
+        url = f"/v2/images?name={name}"
+        logger.info("Gathering image details: %s", name)
+        object_ = self.driver_v2.image_connection.request(url).object
+        images = self.driver_v2._to_images(object_, ex_only_active=False)
+
+        if len(images) != 1:
```

Comment:
In the previous code, we did not care if we found more than one image with the same name. A few years ago I remember the OpenStack image maintainers accidentally uploaded two images with the same name, and obviously that is unlikely to happen often, but we can be robust against that. Let's preserve the existing behavior, so we'll simply pick the first one.

In other words, please change the condition to `if not images:` instead of checking `len() != 1`. That will also match the docstring above that says `ResourceNotFound - when no resource was found`.

Let's also change the exception message to simply be `ResourceNotFound(f"image {name}")`, so users know exactly what type of resource we failed to find.

Reply:
agree... changes have been carried out.